src/index.ts

```diff
@@ -0,0 +1,189 @@
+ * - recordOnlyThisIFrame: FullStory can record the iFrame as its own unique session. Defaults to `false`. Additional conditions apply and can be found [here](https://help.fullstory.com/hc/en-us/articles/360020622514-Can-FullStory-capture-content-that-is-presented-in-iframes-#h_01F1G33B40Q2TPQA8MA7SF8Y5P).
+ * - devMode: In dev mode FullStory won't record sessions. Any calls to SDK methods will `console.warn` that FullStory is in `devMode`. Defaults to `false`.
+ */
+export interface SnippetOptions {
+  orgId: string;
+  namespace?: string;
+  debug?: boolean;
+  host?: string;
+  script?: string;
+  cookieDomain?: string;
+  recordCrossDomainIFrames?: boolean;
+  recordOnlyThisIFrame?: boolean;
+  devMode?: boolean;
+}
+
+/**
+ * A callback that will be invoked when FullStory has begun a session.
+ *
+ * `sessionUrl` contains the URL to the current session.
+ * `settings`   contains the org settings for the current session.
+ */
+type ReadyCallback = (data: { sessionUrl: string, settings: Readonly<object> }) => void;
+
+declare global {
+  interface Window {
+    _fs_cookie_domain?: string;
+    _fs_debug?: boolean;
+    _fs_dev_mode?: boolean;
+    _fs_host?: string;
+    _fs_initialized?: boolean;
+    _fs_is_outer_script?: boolean;
+    _fs_namespace?: string;
+    _fs_org?: string;
+    _fs_run_in_iframe?: boolean;
+    _fs_script?: string;
+  }
+}
+
+const getFullStory = (): FSApi | undefined => {
+  if (window._fs_namespace) {
+    return window[window._fs_namespace];
+  }
+  return undefined;
+};
+
+const ensureSnippetLoaded = (): FSApi => {
+  const fs = getFullStory();
+  if (!fs) {
+    throw Error(
+      'FullStory is not loaded, please ensure the init function is invoked before calling FullStory API functions'
+    );
+  }
+
+  return fs;
+};
+
+const _init = (inputOptions: SnippetOptions, readyCallback?: ReadyCallback) => {
+  // Make a copy so we can modify `options` if desired.
+  const options = { ...inputOptions };
+  if (getFullStory()) {
+    console.warn('The FullStory snippet has already been defined elsewhere (likely in the <head> element)');
+    return;
+  }
+
+  // see README for details on the recordCrossDomainIFrames option
+  if (options.recordCrossDomainIFrames) {
+    window._fs_run_in_iframe = true;
+  }
+
+  // record the contents of this iFrame when embedded in a parent site
+  if (options.recordOnlyThisIFrame) {
+    window._fs_is_outer_script = true;
+  }
+
+  // Set cookie domain if it was specified.
+  if (options.cookieDomain) {
+    window._fs_cookie_domain = options.cookieDomain;
+  }
+
+  if (options.debug === true) {
+    if (!options.script) {
+      options.script = 'edge.fullstory.com/s/fs-debug.js';
+    } else {
+      console.warn('Ignoring `debug = true` because `script` is set');
+    }
+  }
+
+  initFS(options);
+
+  const fs = getFullStory();
+
+  if (!fs) {
+    console.warn('Trouble with fs');
```

Comment:
small nit: seems like `initFS` should have thrown before we get here if anything went wrong, so probably not a big deal:
```suggestion
    console.warn('Failed to initialize FS snippet');
```

Reply:
Oof, yeah I kinda half-baked that log...

I think this is one of those "make typescript happy" if blocks that shouldn't happen, but I'll fix!